kotlinx.fuzz.engine/src/main/kotlin/kotlinx/fuzz/KFuzzerImpl.kt

```diff
@@ -62,20 +62,49 @@ class KFuzzerImpl(data: ByteArray) : KFuzzer, Random() {
         }
     }
 
-    private fun fitIntoRange(
-        value: BigDecimal,
-        oldMin: BigDecimal,
-        oldMax: BigDecimal,
-        newMin: BigDecimal,
-        newMax: BigDecimal,
-    ): BigDecimal {
-        val normalized = value.subtract(oldMin, MathContext.DECIMAL128)
-            .divide(oldMax.subtract(oldMin, MathContext.DECIMAL128), MathContext.DECIMAL128)
-
-        return newMin.add(
-            normalized.multiply(newMax.subtract(newMin, MathContext.DECIMAL128), MathContext.DECIMAL128),
-            MathContext.DECIMAL128,
-        )
+    private inline fun<reified T : Comparable<T>> fitIntoRange(
```

Comment:
One thought: if basically every line of this function uses
```
when (value) {
    is Float -> ...
    is Double -> ...
    else -> ...
}
```

Maybe we should just bite the bullet and split this function into two separate ones? `fitIntoFloatRange` and `fitIntoDoubleRange`. And leave the comment that the implementations are similar in case of future changes

Reply:
Fixed